llvmlite/ir/values.py

```diff
@@ -14,13 +15,15 @@
 
 _VALID_CHARS = (frozenset(map(ord, string.ascii_letters)) |
                 frozenset(map(ord, string.digits)) |
-                frozenset('._-$'))
+                frozenset(map(ord, ' !#$%&\'()*+,-./:;<=>?@[]^_`{|}~')))
 
 
 def _escape_string(text, _map={}):
     """
     Escape the given bytestring for safe use as a LLVM array constant.
     """
+    if isinstance(text, six.text_type):
```

Comment:
Passing an unicode string as an array constant feels like a programmer error to me, so I'd rather raise an exception. I don't know about metadata strings.